jobrunner/record_stats.py

```diff
@@ -17,12 +21,98 @@
 log = logging.getLogger(__name__)
 tracer = trace.get_tracer("ticks")
 
+# Simplest possible table. We're only storing aggregate data
+DDL = """
+CREATE TABLE IF NOT EXISTS jobs (
+    id TEXT,
+    metrics TEXT,
+    PRIMARY KEY (id)
+)
+"""
+
+CONNECTION_CACHE = threading.local()
+
+
+def get_connection(readonly=True):
+    db_file = config.METRICS_FILE
+
+    # developer check against using memory dbs, which cannot be used with this
+    # function, as we need to set mode ourselves
+    assert not db_file.startswith(
```

Comment:
The default `METRICS_FILE` value is a `pathlib.Path` instance and so won't have a `startswith` method. Cast to `str` maybe? Or mabye assert that it's a `pathlib.Path` rather than a `str`?

Reply:
Good catch, fixed.